src/account-info/AccountDetails.js

```diff
@@ -1,59 +1,145 @@
 /* @flow */
 import React, { Component } from 'react';
-import { StyleSheet, Text, View } from 'react-native';
+import { StyleSheet, Text, View, Dimensions } from 'react-native';
 
 import { Avatar, ZulipButton } from '../common';
 import { privateNarrow } from '../utils/narrow';
+import UserStatusIndicator from '../common/UserStatusIndicator';
+
+const ORIENTATION_PORTRAIT = 'PORTRAIT';
+const ORIENTATION_LANDSCAPE = 'LANDSCAPE';
+
+let screenWidth = Dimensions.get('window').width;
 
 const styles = StyleSheet.create({
-  avatarWrapper: {
-    alignItems: 'center',
+  containerLandscape: {
+    flex: 1,
+    flexDirection: 'row'
+  },
+  containerPortrait: {
+    flex: 1,
+    flexDirection: 'column'
+  },
+  avatarContainerLandscape: {
+    flex: 0.5
+  },
+  infoContainer: {
+    padding: 8
+  },
+  infoContainerLandscape: {
+    flex: 0.5
   },
   info: {
     textAlign: 'center',
     fontSize: 18,
   },
   details: {
+    flexDirection: 'row',
+    justifyContent: 'center',
+    alignItems: 'center',
     padding: 8,
   },
-  padding: {
-    padding: 8,
+  statusIndicator: {
+    width: 20,
+    height: 20,
+    marginRight: 5
   },
+  sendButton: {
+    marginRight: 8,
+    marginLeft: 8
+  }
 });
 
 export default class AccountDetails extends Component {
 
+  state = {
+    layoutStyle: ORIENTATION_PORTRAIT
+  };
+  
   handleChatPress = () => {
     const { email, doNarrow, popRoute } = this.props;
     doNarrow(privateNarrow(email));
     popRoute();
   };
 
-  render() {
-    const { avatarUrl, fullName, email, status, auth } = this.props;
+  handleOrientationChange = (event) => {
+    let currentOrientation = '';
+    let screenWidth = Dimensions.get('window').width,
+      screenHeight = Dimensions.get('window').height;
+
+    if (screenWidth > screenHeight) currentOrientation = ORIENTATION_LANDSCAPE;
```

Comment:
Added this logic because `orientation` through redux state was being very unreliable. Let me know if this is a correct thing to do